platform-sdk/swirlds-config-impl/src/main/java/com/swirlds/config/impl/internal/ConfigurationImpl.java

```diff
@@ -110,29 +121,40 @@ public String getValue(@NonNull final String propertyName, @Nullable final Strin
     @Override
     public List<String> getValues(@NonNull final String propertyName) {
         ArgumentUtils.throwArgBlank(propertyName, "propertyName");
-        final String rawValue = getValue(propertyName);
-        if (rawValue == null) {
-            return null;
+        if (!propertiesService.isListProperty(propertyName)) {
+            throw new NoSuchElementException("Property '" + propertyName + "' is not a list property");
         }
-        return ConfigListUtils.createList(rawValue);
+        if (!exists(propertyName)) {
+            throw new NoSuchElementException("Property '" + propertyName + "' not defined in configuration");
+        }
+        return propertiesService.getListProperty(propertyName);
     }
 
     @Nullable
     @Override
     public List<String> getValues(@NonNull final String propertyName, @Nullable final List<String> defaultValue) {
         ArgumentUtils.throwArgBlank(propertyName, "propertyName");
+        if (!propertiesService.isListProperty(propertyName)) {
+            throw new NoSuchElementException("Property '" + propertyName + "' is not a list property");
+        }
         if (!exists(propertyName)) {
             return defaultValue;
         }
-        return getValues(propertyName);
+        return propertiesService.getListProperty(propertyName);
     }
 
     @Nullable
     @Override
     public <T> List<T> getValues(@NonNull final String propertyName, @NonNull final Class<T> propertyType) {
         ArgumentUtils.throwArgBlank(propertyName, "propertyName");
         Objects.requireNonNull(propertyType, "propertyType must not be null");
-        final List<String> values = getValues(propertyName);
+        final List<String> values;
+        if (!propertiesService.isListProperty(propertyName)) {
+            final String value = getValue(propertyName);
+            values = ConfigListUtils.createList(value);
```

Comment:
Do we still want that? My assumption would be to throw an exception if it is not a list.

Reply:
Will be stored as Json Array

Reply:
https://github.com/hashgraph/hedera-services/issues/17102